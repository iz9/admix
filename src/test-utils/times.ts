export function times(amount: number, cb: (...args: any) => any) {
  let i = amount
  while (i > 0) {
    !void cb() && --i
  }
}
