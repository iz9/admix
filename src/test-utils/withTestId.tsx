import React from 'react'

interface WithTestID {
  testId: string
}

export function withTestId<P>(Component: React.ComponentType<P>) {
  const WithTestId: React.FC<P & WithTestID> = props => {
    const { testId, ...rest } = props
    return <Component data-testid={testId} {...(rest as P)} />
  }
  return WithTestId
}
