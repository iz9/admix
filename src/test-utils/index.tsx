import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { withTestId } from './withTestId'
import { ThemeProvider } from '../ui-lib/theme'
import { hexToRGB } from './hexToRgb'

const AllTheProviders: React.FC = ({ children }) => (
  <ThemeProvider>{children as React.ReactChild}</ThemeProvider>
)

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render, withTestId, hexToRGB }
