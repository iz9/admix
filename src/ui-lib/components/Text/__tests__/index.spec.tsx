import React from 'react'

import { Text } from '..'
import { render, screen, cleanup, withTestId } from 'test-utils'

describe('<Text  />', () => {
  afterEach(() => {
    cleanup()
  })
  it('should match snapshot', () => {
    const primaryID = '0'
    const inputID = '1'
    const tagID = '2'
    const titleID = '3'
    const bannerID = '4'
    const Cmp = withTestId(Text)
    render(
      <>
        <Cmp testId={primaryID} variant="primary">
          text
        </Cmp>
        <Cmp testId={inputID} variant="input">
          text
        </Cmp>
        <Cmp testId={tagID} variant="tag">
          text
        </Cmp>
        <Cmp testId={titleID} variant="title">
          text
        </Cmp>
        <Cmp testId={bannerID} variant="banner">
          text
        </Cmp>
      </>,
    )
    expect(screen.getByTestId(primaryID)).toMatchSnapshot()
    expect(screen.getByTestId(inputID)).toMatchSnapshot()
    expect(screen.getByTestId(tagID)).toMatchSnapshot()
    expect(screen.getByTestId(titleID)).toMatchSnapshot()
    expect(screen.getByTestId(bannerID)).toMatchSnapshot()
  })

  it('should render children text ', () => {
    const text = 'text'
    render(<Text color={'black'}>{text}</Text>)
    expect(screen.getByText(text).innerHTML).toBe(text)
  })
})
