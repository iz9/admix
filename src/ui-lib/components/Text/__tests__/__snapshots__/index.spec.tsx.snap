// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text  /> should match snapshot 1`] = `
<span
  class="css-l20krr-Text"
  data-testid="0"
>
  text
</span>
`;

exports[`<Text  /> should match snapshot 2`] = `
<span
  class="css-15cv6um-Text"
  data-testid="1"
>
  text
</span>
`;

exports[`<Text  /> should match snapshot 3`] = `
<span
  class="css-xkjh4q-Text"
  data-testid="2"
>
  text
</span>
`;

exports[`<Text  /> should match snapshot 4`] = `
<span
  class="css-gqz2mm-Text"
  data-testid="3"
>
  text
</span>
`;

exports[`<Text  /> should match snapshot 5`] = `
<span
  class="css-102bbdx-Text"
  data-testid="4"
>
  text
</span>
`;
